clang-tools-extra/clang-tidy/llvmlibc/ImplementationInNamespaceCheck.cpp

```diff
@@ -30,15 +32,18 @@ void ImplementationInNamespaceCheck::check(
     return;
 
   if (const auto *NS = dyn_cast<NamespaceDecl>(MatchedDecl)) {
-    if (NS->getName() != RequiredNamespace) {
-      diag(NS->getLocation(), "'%0' needs to be the outermost namespace")
-          << RequiredNamespace;
-    }
+    if (!Result.SourceManager->isMacroBodyExpansion(NS->getLocation()))
+      diag(NS->getLocation(),
+           "the outermost namespace should be the '%0' macro")
+          << RequiredNamespaceMacroName;
+    else if (!NS->getName().starts_with(RequiredNamespaceStart))
```

Comment:
getName will assert if Namespace got not identifier. What about anonymous namespaces ? Or declarations without name ?